src/liboslexec/runtimeoptimize.cpp

```diff
@@ -2718,7 +2719,10 @@ RuntimeOptimizer::coalesce_temporaries ()
                 if (coalescable (*t) &&
                       equivalent (s->typespec(), t->typespec()) &&
                       s->has_derivs() == t->has_derivs() &&
-                      (slast < t->firstuse() || sfirst > t->lastuse())) {
+                      (slast < t->firstuse() || sfirst > t->lastuse()) &&
+                      ( (!m_opt_batched_analysis) ||
```

Comment:
Just curious... is this test `!m_opt_batched_analysis` necessary? If the analysis is not done, then all the symbols will match each other in their is_uniform and forced_llvm_bool, right? Those will never be different without batched analysis, correct?

Reply:
Currently, if m_opt_batched_analysis is false, then is_uniform == true && forced_llvm_bool == false for all symbols, so you are correct there.  However as analysis of forcing boolean was originally done independent of the uniform vs varying analysis, and theoretically could be used by scalar code I thought it would be useful to call  out that the extra restrictions are only necessary to support the batched code generator.

Reply:
OK, we can drop the "(!m_opt_batched_analysis) ||" portion, as the 2 flags being checked should be set properly to function during scalar and we aren't really saving much by skipping checking to 2 symbol flags vs. 1 global flag.